### PR TITLE
Revert "Improve heuristics whether `format_args` string is a source literal"

### PR DESCRIPTION
**Pull request by Noratrieb**
This reverts commit e6c02aad9345925cfed74f86b414c4d0715d381b (from #106195).

Keeps the code improvements from the PR and the test (as a known-bug).

Works around #106408 while a proper fix is discussed more thoroughly in #106505, as proposed by @tmandry.

Reopens #106191

r? compiler-errors

### CODE REVIEW

**Comment by Nilstrieb** [comment]
I manually tested that this does indeed fix the `indoc` issue.

File: tests/ui/fmt/respanned-literal-issue-106191.rs [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+// unset-rustc-env:RUST_BACKTRACE
+// had to be reverted
+// error-pattern:internal compiler error
+// failure-status:101
+// normalize-stderr-test "DefId([^)]*)" -> "..."
+// normalize-stderr-test "\nerror: internal compiler error.*\n\n" -> ""
+// normalize-stderr-test "note:.*unexpectedly panicked.*\n\n" -> ""
+// normalize-stderr-test "note: we would appreciate a bug report.*\n\n" -> ""
+// normalize-stderr-test "note: compiler flags.*\n\n" -> ""
+// normalize-stderr-test "note: rustc.*running on.*\n\n" -> ""
```
Comment (Nilstrieb) [inline tests/ui/fmt/respanned-literal-issue-106191.rs:13]: i just copied this from the `chalkify/bugs/asyncify.rs` test

Comment (compiler-errors) [reply tests/ui/fmt/respanned-literal-issue-106191.rs:13]: lol

**Comment by compiler-errors** [comment]
r=me once ci is green

@bors p=1 since this should probably make the beta cutoff tomorrow

**Pull request merged**